sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/StorageApiWriteUnshardedRecords.java

```diff
@@ -275,7 +321,13 @@ void flush(RetryManager<AppendRowsResponse, Context<AppendRowsResponse>> retryMa
                   offset = this.currentOffset;
                   this.currentOffset += inserts.getSerializedRowsCount();
                 }
-                return writeStream.appendRows(offset, protoRows);
+                ApiFuture<AppendRowsResponse> response = writeStream.appendRows(offset, protoRows);
+                inflightWaitSecondsDistribution.update(writeStream.getInflightWaitSeconds());
+                if (writeStream.getInflightWaitSeconds() > 5) {
+                  LOG.warn(
+                      "Storage Api write delay more than " + writeStream.getInflightWaitSeconds());
+                }
+                return response;
```

Comment:
would it make sense to add a Distribution here to measure the exec time of this lambda? 
we are locking all the finished bundles on getStreamAppendClient call, maybe we can corroborate/discard any locking delays on high volume there.

Reply:
getInFlightWaitSeconds does measure the exec time of the lambda for the most part (the rest of the code merely appends to an in-memory list, which should be very fast). Locking is better measured with a contention metric I think - need to see if we have any way of measuring this in the wild.